R/00-setup_data.R

```diff
@@ -42,14 +45,14 @@ setup_data <- function(counts = NULL, pg_metadata = NULL, sample_metadata = NULL
 
   new_data$metadata$sample_metadata <- sample_metadata
 
-  # If they don't give paired guide metadata, then we will make up a row id
-  if (is.null(pg_metadata)) pg_metadata <- data.frame(id = 1:ncol(counts))
-  if (!is.data.frame(pg_metadata)) stop("metadata can only be in the form of a data.frame")
-  if (nrow(pg_metadata) != nrow(counts)) stop("the number of rows in the pg_info is not equal to the number of rows in the counts" )
+  # If they don't give paired guide ids, then we will make up a row id
+  if (is.null(pg_ids)) pg_ids <- data.frame(id = 1:ncol(counts))
```

Comment:
I added `pg_ids` so that I know which ID corresponds to which line in the count data and will be putting `pg_metadata` back in once I add the mapping and labeling of constructs as negative control, positive control, single knock out, double knock out. `pg_metadata` made more sense as a variable name for the construct information beyond just the ID itself, but I also don't want to assume that the `pg_metadata` and the count data are sorted the same way

Reply:
But is there a reason the checks for the metadata are no longer here?